CoreFoundation/Base.subproj/CFKnownLocations.c

```diff
@@ -94,21 +95,15 @@ CFURLRef _Nullable _CFKnownLocationCreatePreferencesURLForUser(CFKnownLocationUs
 #elif TARGET_OS_ANDROID
 
     switch (user) {
-    case _kCFKnownLocationUserAny:
-    case _kCFKnownLocationUserByName:
-      abort();
-    case _kCFKnownLocationUserCurrent: {
-      const char *buffer = getenv("CFFIXED_USER_HOME");
-      if (buffer == NULL || *buffer == '\0') {
-        CFLog(__kCFLogAssertion, CFSTR("CFFIXED_USER_HOME is unset"));
-        HALT;
-      }
-
-      CFURLRef userdir = CFURLCreateFromFileSystemRepresentation(kCFAllocatorSystemDefault, (const unsigned char *)buffer, strlen(buffer), true);
-      location = CFURLCreateWithFileSystemPathRelativeToBase(kCFAllocatorSystemDefault, CFSTR("/Apple/Library/Preferences"), kCFURLPOSIXPathStyle, true, userdir);
-      CFRelease(userdir);
-      break;
-    }
+        case _kCFKnownLocationUserAny:
+        case _kCFKnownLocationUserByName:
```

Comment:
I think that this should still `abort`.  `CFCopyHomeDirectoryURL` will give you the current user's home directory.

Reply:
Please, read the explanation in the commit message. There's no users in Android. The sandboxing only allows the app to write to their sandbox. Any User, Current User, and User By Name are effectively the same. The app should set a `HOME` (or `CFFIXED_USER_HOME`) to point at some point in their sandbox. There's no going to be preferences for “any user” beside the current user, so trying to choose different directories is not worth it.

Besides, the `abort` in the previous code was making the test suite abort, because the test suite uses `Any` (through `UserPreferences`), and it is not possible to avoid so.

Reply:
I did read the explanation, but that wasn't particularly clear from the commit message.

Reply:
Do you want me to rewrite the commit message/PR description and resubmit? Or should I change something else?

We need at least `Any` and `UserCurrent` to work for the test suite to pass without Android-specific changes.